### PR TITLE
Test room visibility changes

### PR DESCRIPTION
**Pull request by babolivier**
Based off #3 

Tests to make sure changes in access rules and in visibility in the server's room directory interact correctly.

### CODE REVIEW

File: room_access_rules/__init__.py [hunk 1]
```diff
@@ -393,10 +393,7 @@ async def check_event_allowed(
...
         self, room_id: str, state_events: StateMap[EventBase], new_visibility: str
     ) -> bool:
-        """Implements
-        synapse.events.ThirdPartyEventRules.check_visibility_can_be_modified
-
-        Determines whether a room can be published, or removed from, the public room
+        """Determines whether a room can be published, or removed from, the public room
         list. A room is published if its visibility is set to "public". Otherwise,
         its visibility is "private". A room with access rule other than "restricted"
         may not be published.
```
Comment (DMRobertson) [inline room_access_rules/__init__.py:399]: I've seen other code that speaks of "world readable" rooms. Is that a related or relevant concept here?

File: tests/test_visibility.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        'restricted' when the room is published in the room directory.
+        """
+        state = {
+            (ACCESS_RULES_TYPE, ""): new_access_rules_event(
+                self.user_id, PUBLIC_ROOM_ID, AccessRules.RESTRICTED,
+            )
+        }
+
+        # Check that we can't change the rule to 'unrestricted'.
+        allowed, _ = await self.module.check_event_allowed(
```
Comment (DMRobertson) [inline tests/test_visibility.py:38]: Which function are we calling here---this one?

https://github.com/vector-im/synapse-room-access-rules/blob/084ba5b6c986680dfab117b6bc869f8377b0b508/room_access_rules/__init__.py#L335-L339

The return type annotation for that says it returns either a `bool` or a `dict`. We seem to be expecting a Tuple here of two things and discarding the second, so I don't understand what's going on here.

Comment (DMRobertson) [reply tests/test_visibility.py:38]: Oh sorry, this depends on #3. Let me take a look at that now.

Comment (babolivier) [reply tests/test_visibility.py:38]: Yeah this callback wasn't ported correctly and had the old return type/values, I fixed it in #3

**Review by DMRobertson — APPROVED** [review]
LGTM!
